lib/rubocop/cop/sorbet/forbid_include_const_literal.rb

```diff
@@ -26,41 +26,44 @@ module Sorbet
       # ```ruby
       # include Polaris::Engine.helpers
       # ```
-      class ForbidIncludeConstLiteral < RuboCop::Cop::Cop # rubocop:todo InternalAffairs/InheritDeprecatedCopClass
-        MSG = "Includes must only contain constant literals"
+      class ForbidIncludeConstLiteral < RuboCop::Cop::Base
+        extend AutoCorrector
 
-        attr_accessor :used_names
+        MSG = "`%<inclusion_method>s` must only be used with constant literals as arguments"
+        RESTRICT_ON_SEND = [:include, :extend, :prepend].freeze
 
-        # @!method not_lit_const_include?(node)
-        def_node_matcher :not_lit_const_include?, <<-PATTERN
-          (send nil? {:include :extend :prepend}
-            $_
-          )
+        # @!method dynamic_inclusion?(node)
+        def_node_matcher :dynamic_inclusion?, <<~PATTERN
+          (send $_ ${:include :extend :prepend} $#neither_const_nor_self?)
         PATTERN
 
-        def initialize(*)
-          super
-          self.used_names = Set.new
-        end
+        def_node_matcher :t_unsafe?, <<~PATTERN
+          (send (const {nil? cbase} :T) :unsafe _)
+        PATTERN
 
         def on_send(node)
-          return unless not_lit_const_include?(node) do |send_argument|
-            ![:const, :self].include?(send_argument.type)
+          dynamic_inclusion?(node) do |receiver, inclusion_method, included|
+            return if t_unsafe?(receiver)
+            return unless within_onymous_module?(node)
+
+            add_offense(node, message: format(MSG, inclusion_method: inclusion_method)) do |corrector|
+              explicit_receiver = receiver&.source || 'self'
+              corrector.replace(node, "T.unsafe(#{explicit_receiver}).#{inclusion_method} #{included.source}")
+            end
           end
-          parent = node.parent
-          return unless parent
-          parent = parent.parent if [:begin, :block].include?(parent.type)
-          return unless [:module, :class, :sclass].include?(parent.type)
-          add_offense(node)
         end
 
-        def autocorrect(node)
-          lambda do |corrector|
-            corrector.replace(
-              node.source_range,
-              "T.unsafe(self).#{node.source}"
-            )
-          end
+        private
+
+        def neither_const_nor_self?(node)
+          !node.const_type? && !node.self_type?
+        end
+
+        # Returns true if the node is within a module declaration that is not anonymous.
+        def within_onymous_module?(node)
```

Comment:
Do you mean:
```suggestion
        def within_anonymous_module?(node)
```

Reply:
I think we can tune down on the sesquipedalianism and keep the codebase easy to understand without a dictionary. 

`within_named_module?` would have worked.